packages/expo/src/Notifications.ts

```
@@ -244,13 +312,28 @@ export default {
     }
   },
 
+  async scheduleLocalNotificationWithMatchAsync(notification, options) {
+    if (Platform.OS === 'android') {
+       return new Promise((resolve,reject) => reject("This function currently works only on ios"));
```

You could just throw an Error
```
throw new Error("This function is supported only on iOS.")
```